### PR TITLE
Add test framework for gateway

### PR DESCRIPTION
**Pull request by willmostly**
Add a test script for the gateway mimicking the form of the trino tests and call it in `ci-cd`. More comprehensive test logic will be added in follow up PRs

### CODE REVIEW

**Comment by willmostly** [comment]
@nineinchnick do you prefer a separate test script as done here or should I merge this into the existing `test.sh`?

File: gateway-tests/test-values.yaml [hunk 1]
```diff
@@ -1,7 +1,8 @@
 replicaCount: 1
 
 image:
-  repository: "trino-gateway"
+  # -- Repository location of the Trino Gateway image, typically `organization/imagename`
```
Comment (nineinchnick) [inline gateway-tests/test-values.yaml:4]: Move this to the previous commit.

File: .github/workflows/ci-cd.yaml [hunk 2]
```diff
@@ -26,7 +26,7 @@ jobs:
       - name: Set up chart-testing
         uses: helm/chart-testing-action@v2.6.1
       - name: Lint charts
-        run: ct lint --charts=charts/trino --validate-maintainers=false
+        run: ct lint --charts=charts/trino,charts/gateway --validate-maintainers=false
```
Comment (nineinchnick) [inline .github/workflows/ci-cd.yaml:29]: Move this into a separate commit - the next one that also fixes the linting issues.

File: gateway-tests/test.sh [hunk 3]
```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+declare -A testCases=(
+    [complete_values]="--values test-values.yaml"
+)
+
+declare -A testCaseCharts=(
+    [complete_values]="../charts/gateway"
```
[the remark this answers is not in the document]
Comment (nineinchnick) [reply gateway-tests/test.sh:10]: Also, let's move all tests into a `test` directory first.

[the remark this answers is not in the document]
Comment (willmostly) [reply gateway-tests/test.sh]: I created `tests` and moved the gateway tests in - if that looks good to you i'll move the trino tests in as well.

Reusing the DB and any other common fixtures seems like a good idea. I considered it, but wanted to avoid conflicts while getting things working. Lets do it as a follow up

File: .github/workflows/ci-cd.yaml [hunk 4]
```diff
@@ -55,8 +54,31 @@ jobs:
         uses: helm/kind-action@v1.10.0
         with:
           version: v0.24.0
-      - name: Run tests
-        run: ./test.sh ${{ matrix.args }}
+      - name: Run trino tests
```
Comment (nineinchnick) [inline .github/workflows/ci-cd.yaml:57]: Keep this step name as `Run tests`, but change the job name on line 33 to `trino-test`, so it is distinguishable from gateway tests.

File: .github/workflows/ci-cd.yaml [hunk 5]
```diff
@@ -55,8 +54,31 @@ jobs:
...
+        run: ./tests/trino/test.sh ${{ matrix.args }}
+
+  test-gateway:
+    runs-on: ubuntu-latest
+    name: test ${{ matrix.label }}
```
Comment (nineinchnick) [inline .github/workflows/ci-cd.yaml:62]:
```suggestion
    name: test gateway ${{ matrix.label }}
```

**Review by nineinchnick — APPROVED** [review]
LGTM after adjusting CI job names.

File: .github/workflows/ci-cd.yaml [hunk 6]
```diff
@@ -26,11 +26,11 @@ jobs:
       - name: Set up chart-testing
         uses: helm/chart-testing-action@v2.6.1
       - name: Lint charts
-        run: ct lint --charts=charts/trino --validate-maintainers=false
+        run: ct lint --charts=charts/trino,charts/gateway --validate-maintainers=false
 
   test:
     runs-on: ubuntu-latest
-    name: test ${{ matrix.label }}
+    name: trino-test ${{ matrix.label }}
```
Comment (nineinchnick) [inline .github/workflows/ci-cd.yaml:33]: Make it consistent with the `test gateway` job:
```suggestion
    name: test trino ${{ matrix.label }}
```

**Review by nineinchnick — APPROVED** [review]
